src/boot_O2_g3/boot_main.c

```diff
@@ -3,19 +3,19 @@
 
 StackEntry sBootThreadInfo;
 OSThread sIdleThread;
-u8 sIdleThreadStack[0x400];
+STACK(sIdleThreadStack, 0x400);
 StackEntry sIdleThreadInfo;
-u8 sBootThreadStack[0x400];
+STACK(sBootThreadStack, 0x400);
 
 void bootproc(void) {
-    StackCheck_Init(&sBootThreadInfo, sBootThreadStack, sBootThreadStack + sizeof(sBootThreadStack), 0, -1, "boot");
+    StackCheck_Init(&sBootThreadInfo, sBootThreadStack, STACK_TOP(sBootThreadStack), 0, -1, "boot");
     osMemSize = osGetMemSize();
     func_800818F4();
     osInitialize();
     osUnmapTLBAll();
     gCartHandle = osCartRomInit();
-    StackCheck_Init(&sIdleThreadInfo, sIdleThreadStack, sIdleThreadStack + sizeof(sIdleThreadStack), 0, 256, "idle");
-    osCreateThread(&sIdleThread, Z_THREAD_ID_IDLE, Idle_ThreadEntry, NULL, sIdleThreadStack + sizeof(sIdleThreadStack),
+    StackCheck_Init(&sIdleThreadInfo, sIdleThreadStack, STACK_TOP(sIdleThreadStack), 0, 256, "idle");
```

Comment:
```suggestion
    StackCheck_Init(&sIdleThreadInfo, sIdleThreadStack, STACK_TOP(sIdleThreadStack), 0, 0x100, "idle");
```
Every other one uses hex.